Traceback (most recent call last):
  File "/home/bitzer/SoftDev/Python/ScienceEnv/SE/lib/python3.9/site-packages/jupyter_cache/executors/utils.py", line 51, in single_nb_execution
    executenb(
  File "/home/bitzer/SoftDev/Python/ScienceEnv/SE/lib/python3.9/site-packages/nbclient/client.py", line 1117, in execute
    return NotebookClient(nb=nb, resources=resources, km=km, **kwargs).execute()
  File "/home/bitzer/SoftDev/Python/ScienceEnv/SE/lib/python3.9/site-packages/nbclient/util.py", line 78, in wrapped
    return just_run(coro(*args, **kwargs))
  File "/home/bitzer/SoftDev/Python/ScienceEnv/SE/lib/python3.9/site-packages/nbclient/util.py", line 57, in just_run
    return loop.run_until_complete(coro)
  File "/usr/lib/python3.9/asyncio/base_events.py", line 642, in run_until_complete
    return future.result()
  File "/home/bitzer/SoftDev/Python/ScienceEnv/SE/lib/python3.9/site-packages/nbclient/client.py", line 558, in async_execute
    await self.async_execute_cell(
  File "/home/bitzer/SoftDev/Python/ScienceEnv/SE/lib/python3.9/site-packages/nbclient/client.py", line 862, in async_execute_cell
    self._check_raise_for_error(cell, exec_reply)
  File "/home/bitzer/SoftDev/Python/ScienceEnv/SE/lib/python3.9/site-packages/nbclient/client.py", line 765, in _check_raise_for_error
    raise CellExecutionError.from_cell_and_msg(cell, exec_reply_content)
nbclient.exceptions.CellExecutionError: An error occurred while executing the following cell:
------------------
import matplotlib
import numpy
from matplotlib import pyplot
from myst_nb import glue

matplotlib.style.use('sv.mplstyle')

num_samples = 16
f0 = 155
fs = 1000

sample = [0]*num_samples

kk = numpy.arange(num_samples)
cos_series = numpy.cos(2*numpy.pi*f0*kk/fs)

sample = numpy.concatenate(([0]*4,cos_series,[0]*4))

reconst_sig = numpy.concatenate((cos_series[-4:],cos_series,cos_series[0:4]))


# signal reconstruction from fft through ifft
spectrum = numpy.fft.fft(sample,256)
spectrumshort = numpy.fft.fft(sample)

spectrumshort = numpy.fft.fftshift(numpy.abs(spectrumshort)/numpy.abs(spectrumshort).max())
spectrum = numpy.fft.fftshift(spectrum)
# dB normalized to 0 max amplitude
spectrum_abs = numpy.abs(spectrum)/numpy.abs(spectrum).max() 
# frequency bins for plotting
freqs_fft = numpy.linspace(-1, 1, len(spectrum_abs)) 
freqs_fftshort = numpy.linspace(-1, 1, len(spectrumshort)) 


fig, ax = pyplot.subplots(2, 2)
print(ax)
ax[0][0].stem(range(-4,len(sample)-4),sample)
ax[0][0].set(xlabel='Folgenndex k ->', ylabel='Amplitude x[k]', 
        title='Eingangssignal Sinus')

ax[0][1].plot(freqs_fft, spectrum_abs)
ax[0][1].set(xlabel='Frequenz rad/pi ', 
        ylabel='Frequenzanteil normiert auf 1', 
        title='Amplitudenspektrum der gefensterten Cosinus-Fkt')

# f2, ax_spec_stem, ax_ifft) = pyplot.subplots(1, 2)
ax[1][0].stem(freqs_fftshort, spectrumshort, use_line_collection=True)
ax[1][0].set(xlabel='Frequenz rad/pi ', 
        ylabel='Frequenzanteil normiert auf 1',
        title='Abgetastetes Amplitudenspektrum')

ax[1][1].stem(range(-4,len(sample)-4),reconst_sig)
ax[1][1].set(xlabel='Folgenndex k ->', ylabel='Amplitude x[k]', 
        title='durch Abtastung des Spektrums erzieltes rekonstruiertes Signal')
ax_ifft.plot([-0.5,-0.5],[-1,1], 'r:' )
ax_ifft.plot([15.5,15.5],[-1,1], 'r:')

pyplot.tight_layout()

glue("DFT_Cosinus", fig, display=False)

------------------

[0;31m---------------------------------------------------------------------------[0m
[0;31mNameError[0m                                 Traceback (most recent call last)
[0;32m/tmp/ipykernel_101961/605318221.py[0m in [0;36m<module>[0;34m[0m
[1;32m     53[0m ax[1][1].set(xlabel='Folgenndex k ->', ylabel='Amplitude x[k]', 
[1;32m     54[0m         title='durch Abtastung des Spektrums erzieltes rekonstruiertes Signal')
[0;32m---> 55[0;31m [0max_ifft[0m[0;34m.[0m[0mplot[0m[0;34m([0m[0;34m[[0m[0;34m-[0m[0;36m0.5[0m[0;34m,[0m[0;34m-[0m[0;36m0.5[0m[0;34m][0m[0;34m,[0m[0;34m[[0m[0;34m-[0m[0;36m1[0m[0;34m,[0m[0;36m1[0m[0;34m][0m[0;34m,[0m [0;34m'r:'[0m [0;34m)[0m[0;34m[0m[0;34m[0m[0m
[0m[1;32m     56[0m [0max_ifft[0m[0;34m.[0m[0mplot[0m[0;34m([0m[0;34m[[0m[0;36m15.5[0m[0;34m,[0m[0;36m15.5[0m[0;34m][0m[0;34m,[0m[0;34m[[0m[0;34m-[0m[0;36m1[0m[0;34m,[0m[0;36m1[0m[0;34m][0m[0;34m,[0m [0;34m'r:'[0m[0;34m)[0m[0;34m[0m[0;34m[0m[0m
[1;32m     57[0m [0;34m[0m[0m

[0;31mNameError[0m: name 'ax_ifft' is not defined
NameError: name 'ax_ifft' is not defined

